### PR TITLE
Issue 35: Make a recommendation on continuous integration.

### PR DESCRIPTION
**Pull request by trenaman**
For discussion at next Arch Board meeting.

### CODE REVIEW

File: recommendations/ContinuousIntegration.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## Assess
+
+  - Per-department SaaS for closed-source IF there are no internal dependencies.
+
+## Hold
+
+  - A centralized Jenkins system will continue to be maintained for historical reasons, to support ION Cannon, daily performance tests, and Gerrit. This will owned by INFRA. Engineering teams should NOT add new projects into this centralized repository, but should prefer to adopt per-department CI technologies as appropriate.
+
+  - SaaS CI products should not bused for closed-source with internal / private dependencies.
```
Comment (Seanstoppable) [inline recommendations/ContinuousIntegration.md]: bused = be used

Comment (jkenny) [inline recommendations/ContinuousIntegration.md]: "technologies", not "tech nologies"

File: recommendations/ContinuousIntegration.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Continous Integration
+
+Gilt has adopted a number of continuous integration technologies; most notably Jenkins which is used to power ION-Cannon deployment, Gerrit and the Daily Load Tests. A centralized Jenkins system will continue to be maintained for historical reasons, to support ION Cannon, daily performance tests, and Gerrit. This will owned by INFRA. Engineering teams should NOT add new projects into this centralized repository, but should prefer to adopt per-department CI tech nologies as appropriate. While there continues to be innovation in the CI space, there is a general consensus that we do not wish to invest or innovate in this area; teams should be free to use whatever tool is most appropriate to their needs, with a preference to open-source and low-cost options.
+
+## Adopt
+  - Per-department Jenkins instances for closed-source with dependencies on private code within Gilt.
```
Comment (arbjerg) [inline recommendations/ContinuousIntegration.md]: oops, the above is me, I just noticed I was logged under the Data Automation account and not my own...

**Comment by arbjerg** [comment]
LGTM
I created a repo to accellerate this and save teams from doing a lot of dull research; https://github.com/gilt//tools-ci. This repo contains the make-tools and Cloudformation templates to easily setup a Jenkins server in AWS. Also, it includes tools and instructions for setting up AWS Code Pipelines and Code Deploy with Jenkins build steps.

File: recommendations/ContinuousIntegration.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  - Per-department SaaS services (e.g. [Travis](https://travis-ci.org), [CircleCI](https://circleci.com)) for open-source.
```
Comment (k4sbasia) [inline recommendations/ContinuousIntegration.md]: @trenaman remove "for open-source"? In ouroboros we are using circleCI for non open-sourced software hosted on github. I think @kevinoriordan does the same for cerebro projects.

File: recommendations/ContinuousIntegration.md (hunk 2 above)
Comment (trenaman) [inline recommendations/ContinuousIntegration.md]: Sounds good.

File: recommendations/ContinuousIntegration.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Continous Integration
+
+Gilt has adopted a number of continuous integration technologies; most notably Jenkins which is used to power ION-Cannon deployment, Gerrit and the Daily Load Tests. A centralized Jenkins system will continue to be maintained for historical reasons, to support ION Cannon, daily performance tests, and Gerrit. This will owned by INFRA. Engineering teams should NOT add new projects into this centralized repository, but should prefer to adopt per-department CI technologies as appropriate. While there continues to be innovation in the CI space, there is a general consensus that we do not wish to invest or innovate in this area; teams should be free to use whatever tool is most appropriate to their needs.
```
Comment (gheine) [inline recommendations/ContinuousIntegration.md]: Remove repetition as well as "gilt lingo":

Gilt has adopted a number of continuous integration technologies; most notably Jenkins which is used to power legacy deployment, Gerrit and the Daily Load Tests. To facititate a smooth transition,  a centralized Jenkins system will continue to be maintained for historical reasons. Engineering teams should NOT add new projects into this centralized repository, but should prefer to adopt per-department CI technologies as appropriate. While there continues to be innovation in the CI space, there is a general consensus that we do not wish to invest or innovate in this area; teams should be free to use whatever tool is most appropriate to their needs.

File: recommendations/ContinuousIntegration.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+## Adopt
+  - Per-department Jenkins instances for closed-source with dependencies on private code within Gilt.
+
+  - Per-department SaaS services (e.g. [Travis](https://travis-ci.org), [CircleCI](https://circleci.com)) for open-source projects or private repos on github that do not have internal dependencies.
+
+## Hold
+
+  - A centralized Jenkins system will continue to be maintained for historical reasons, to support ION Cannon, daily performance tests, and Gerrit. This will owned by INFRA. Engineering teams should NOT add new projects into this centralized repository, but should prefer to adopt per-department CI technologies as appropriate.
```
Comment (gheine) [inline recommendations/ContinuousIntegration.md]: As above:

A centralized Jenkins system will continue to be maintained for historical reasons, to support legacy deployment, daily performance tests, and Gerrit. Engineering teams should NOT add new projects into this centralized repository, but should prefer to adopt per-department CI technologies as appropriate.

**Pull request merged**